spec/system/module_overview_progress_spec.rb

```diff
@@ -0,0 +1,129 @@
+    before do
+      start_module(alpha)
+      visit '/modules/alpha'
+    end
+
+    it 'first submodule has not been started yet' do
+      within '#section-button-1' do
+        expect(page).to have_content('not started')
+      end
+
+      within '#section-content-1 .govuk-list li:first-child' do
+        expect(page).to have_content('1-1-1')
+          .and have_content('not started')
+        expect(page).not_to have_link('1-1-1')
+      end
+
+      expect(page).to have_link('Resume training', href: '/modules/alpha/content-pages/intro')
+    end
+  end
+
+  context 'when the user has viewed the first submodule page' do
+    before do
+      start_submodule(alpha)
+      visit '/modules/alpha'
+    end
+
+    it 'submodule has been started' do
+      within '#section-button-1' do
+        expect(page).to have_content('in progress')
+      end
+
+      within '#section-content-1 .govuk-list li:first-child' do
+        expect(page).to have_content('not started')
+          .and have_link('1-1-1', href:  '/modules/alpha/content-pages/1-1-1')
+      end
+
+      expect(page).to have_link('Resume training', href: '/modules/alpha/content-pages/1-1')
+    end
+  end
+
+  context 'when the user has viewed the first topic page' do
+    before do
+      start_topic(alpha)
+      visit '/modules/alpha'
+    end
+
+    it 'first topic has been completed' do
+      within '#section-button-1' do
+        expect(page).to have_content('in progress')
+      end
+
+      within '#section-content-1 .govuk-list li:first-child' do
+        expect(page).to have_content('complete')
+      end
+
+      within '#section-content-1 .govuk-list li:nth-child(2)' do
+        expect(page).to have_content('not started')
+          .and have_link('1-1-2', href:  '/modules/alpha/content-pages/1-1-2')
+      end
+
+      expect(page).to have_link('Resume training', href: '/modules/alpha/content-pages/1-1-1')
+    end
+  end
+
+  context 'when the user has finished the first submodule' do
+    before do
+      view_pages_before_formative_assessment(alpha)
+      visit '/modules/alpha'
+    end
+
+    it 'first submodule is complete' do
+      within '#section-button-1' do
+        expect(page).to have_content('complete')
+      end
+
+      within '#section-content-1 .govuk-list' do
+        expect(page).to have_content('complete', count: 4)
+      end
+
+      expect(page).to have_link('Resume training', href: '/modules/alpha/content-pages/1-1-4')
+    end
+  end
+
+  context 'when the user has failed a summative assessment' do
+    before do
+      view_pages_before_summative_assessment(alpha)
+      visit '/modules/alpha/summative-assessments/1-3-2-1'
+      3.times do
+        check 'Wrong answer 1'
+        check 'Wrong answer 2'
+        click_on 'Save and continue'
+      end
+      choose 'Wrong answer 1'
+      click_on 'Finish test'
+      visit '/modules/alpha'
+    end
+
+    it 'button shows option to retake test' do
```

Comment:
`it 'has a button to retake the test'` is right but could potentially pass if that text also appears somewhere else on the page. This starts to fall under our `ui` framework more but I'd be inclined to use the most specific matcher we can i.e. `have_button`

Reply:
At which point, you might like to see what the auto-generated description looks like. For example, if you removed the handwritten description passed to `it`, often this is accompanied with changing `it` to `specifiy`, see what the assertion spits out as a description when you use `-f doc`.